core/store/migrations/migration1565877314/migrate.go

```diff
@@ -0,0 +1,67 @@
+package migration1565877314
+
+import (
+	"github.com/jinzhu/gorm"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+)
+
+type ExternalInitiatorOld struct {
+	*gorm.Model
+	AccessKey    string
+	Salt         string
+	HashedSecret string
+}
+
+func (ExternalInitiatorOld) TableName() string {
+	return "external_initiators"
+}
+
+type ExternalInitiator struct {
+	*gorm.Model
+	Name         string        `gorm:"not null,unique"`
+	URL          models.WebURL `gorm:"not null"`
+	AccessKey    string
+	Salt         string
+	HashedSecret string
+}
+
+// newExternalInitiator creates a new row, setting the Name to the AccessKey
+func newExternalInitiator(arg ExternalInitiatorOld) ExternalInitiator {
+	url, _ := models.NewWebURL("https://unset.url")
+	return ExternalInitiator{
+		Model: arg.Model,
+
+		Name:         arg.AccessKey,
+		URL:          url,
+		AccessKey:    arg.AccessKey,
+		Salt:         arg.Salt,
+		HashedSecret: arg.HashedSecret,
+	}
+}
+
+// Migrate adds External Initiator Name and URL fields.
+func Migrate(tx *gorm.DB) error {
+	var exis []ExternalInitiatorOld
+	if err := tx.Find(&exis).Error; err != nil {
```

Comment:
I'd definitely want to try doing this with SQL rather than loading all values into Go land as this will likely explode at scale.

Reply:
It was under the assumption there's a minimal amount of exi's.  If this is an issue, up for using a temporary table instead of memory?  I opine it is best to stay in the domain of Golang's GORM to avoid issues between Sqlite3 and Postgres.

Reply:
@j16r there should be almost no external initiators as it's still behind the DEV flag, so in 99% of cases not there will be 0 records.